src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -33,17 +33,15 @@ public class ModuleIntrinsics
         [TraceSource("Modules", "Module loading and analysis")]
```

Comment:
@iSazonov, your last commit had 6 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

SxS Module Path Basic Tests.validate sxs module path
```powershell
Expected 3, but got 0.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/engine/Module/ModulePath.Tests.ps1: line 65
65:             $paths.Count | Should -Be 3
```
SxS Module Path Basic Tests.ignore pshome module path derived from a different PowerShell instance
```powershell
Expected 3, but got 1.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/engine/Module/ModulePath.Tests.ps1: line 97
97:                 $paths.Count | Should -Be 3
```
SxS Module Path Basic Tests.keep non-pshome module path derived from PowerShell instance parent
```powershell
Expected 5, but got 2.
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/engine/Module/ModulePath.Tests.ps1: line 130
130:             $paths.Count | Should -Be 5
```
TabCompletion.Should complete abbreviated function
```powershell
Expected exactly 'PSConsoleHostReadLine', but got @('PSConsoleHostReadLine', 'PSReadLine\PSConsoleHostReadLine').
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1: line 22
22:         $res | Should -BeExactly 'PSConsoleHostReadLine'
```
Import-Module.should be able to add a module with using ModuleInfo switch
```powershell
Expected exactly 'Microsoft.PowerShell.Security', but got @('Microsoft.PowerShell.Security', 'Microsoft.PowerShell.Security').
at <ScriptBlock>, /Users/vsts/agent/2.155.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1: line 26
26:         (Get-Module -Name $moduleName).Name | Should -BeExactly $moduleName
```